src/api/controllers/tests/user-controller.test.ts

```diff
@@ -0,0 +1,230 @@
+                expect(res.body[0]).toHaveProperty('phone');
+                expect(res.body[0]).toHaveProperty('email');
+                userRepoSpy.mockRestore();
+            });
+
+            it('should return an empty array.', async() => {
+                //  Arrange
+                const endpoint = '/api/user/getAllUsers';
+                const expectedStatusCode = 200;
+                const userRepoSpy = jest
+                .spyOn(userRepos, 'getAllUsers')
+                .mockResolvedValueOnce([]); 
+
+                //  Act
+                const res = await request(app).get(endpoint)
+
+                //  Assert
+                expect(res.statusCode).toBe(expectedStatusCode);
+                expect(res.body).toEqual([]);
+                expect(userRepoSpy).toHaveBeenCalledWith();
+                expect(userRepoSpy).toHaveBeenCalledTimes(1);
+                userRepoSpy.mockRestore();
+            });
+        });
+        describe('getUserByEmail endpoint', () => {
+            it("should return the correct user.", async () => {
+                //  Arrange
+                const endpoint = '/api/user/getUserByEmail';
+                const testEmail = 'sample2@gmail.com';
+                const expectedStatusCode = 200;
+                const userRepoSpy = jest
+                .spyOn(userRepos, 'getUserByEmail')
+                .mockResolvedValueOnce({
+                    id: 1,
+                    firebase_id: '1fnj3u4hsd',
+                    firstName: 'Lenny',
+                    lastName: 'Jenkins-Joules',
+                    email: 'sample2@gmail.com',
+                    phone: '6146156164',
+                    countryCode: 1,
+            });
+
+                //  Act
+                const res = await request(app)
+                .get(endpoint)
+                .send({
+                    email: testEmail
+                });
+
+                //  Assert
+                expect(res.statusCode).toEqual(expectedStatusCode);
+                expect(userRepoSpy).toHaveBeenCalledTimes(1);
+                expect(userRepoSpy).toHaveBeenCalledWith(testEmail);
+                expect(res.headers['content-type']).toEqual('application/json; charset=utf-8');
+                expect(res.body.id).toEqual(1);
+                expect(res.body.firebase_id).toEqual('1fnj3u4hsd');
+                expect(res.body.firstName).toEqual('Lenny');
+                expect(res.body.lastName).toEqual('Jenkins-Joules');
+                expect(res.body.email).toEqual('sample2@gmail.com');
+                expect(res.body.phone).toEqual('6146156164');
+                expect(res.body.countryCode).toEqual(1);
+                userRepoSpy.mockRestore();
+            });
+
+            it("should return an empty JSON object.", async () => {
+                //  Arrange
+                const endpoint = '/api/user/getUserByEmail';
+                const testEmail = 'invalidEmail';
```

Comment:
This will not be accepted by the schema validation. You are checking for a [valid email string here](https://github.com/Chrysalis-Papilio/Soen-490-Papilio-Backend/blob/3b2cb50cb365e4d1bb628cb63b57f6f0e101c9a6/src/api/schemas/user-schema.ts#L54). You should rework this test to make it work with a valid email to pass the schema validation.
This will return a status code 400 because you don't have the correct information for this endpoint